q2_types/_feature_data.py

```diff
@@ -153,6 +162,26 @@ def generator_to_dna_sequences(view, data_dir):
     skbio.io.write(view, format='fasta', into=file)
 
 
+def paired_dna_sequences_to_generator(data_dir):
+    left = skbio.io.read(os.path.join(data_dir, 'left-dna-sequences.fasta'),
+                         format='fasta', constructor=skbio.DNA)
+    right = skbio.io.read(os.path.join(data_dir, 'right-dna-sequences.fasta'),
+                          format='fasta', constructor=skbio.DNA)
+    for lseq, rseq in zip(left, right):
+        assert rseq.metadata['id'] == lseq.metadata['id'], \
```

Comment:
Can you raise a `ValueError` stating that the sequence identifiers don't match? This error could be user-facing and something we'd want to happen every time (assert statements can be turned off depending on how Python is executed).